plugins/linux-pipewire/pipewire.h

```diff
@@ -22,9 +22,16 @@
 
 #include <obs-module.h>
 
+#include <pipewire/keys.h>
+#include <pipewire/properties.h>
+
 typedef struct _obs_pipewire_data obs_pipewire_data;
 
-void *obs_pipewire_create(int pipewire_fd, int pipewire_node);
+extern const struct pw_stream_events stream_events;
+
+void *obs_pipewire_create(int pipewire_fd, int pipewire_node, const char *name,
```

Comment:
Why is this void*, the only assignment from this function i see is directly into the correct type.

Reply:
The reason for this decision,  i can think of, is to hide all (for the plugin) "unnecessary" stuff behind a opaque handle.

Reply:
Can we use an opaque type instead of void*, unless there is a real reason we need to use void* (but since this always returns the same type it doesnt seem so?).

Reply:
Why not? But I'd like implement this after we finished designing the whole wrapper api.